syntax/entry.php

```diff
@@ -296,18 +298,23 @@ function replaceQuery() {
      * @param Doku_Renderer_plugin_data_edit $renderer
      */
     function _editData($data, &$renderer) {
-        $renderer->form->startFieldset($this->getLang('dataentry'));
-        $renderer->form->_content[count($renderer->form->_content) - 1]['class'] = 'plugin__data';
-        $renderer->form->addHidden('range', '0-0'); // Adora Belle bugfix
+        $renderer->form->addFieldsetOpen($this->getLang('dataentry'));
+        $renderer->form->getElementAt($renderer->form->elementCount() - 1)->attr('class', 'plugin__data');
+        $renderer->form->setHiddenField('range', '0-0'); // Adora Belle bugfix
 
         if($this->getConf('edit_content_only')) {
-            $renderer->form->addHidden('data_edit[classes]', $data['classes']);
+            $renderer->form->setHiddenField('data_edit[classes]', $data['classes']);
 
             $columns = array('title', 'value', 'comment');
             $class = 'edit_content_only';
 
         } else {
-            $renderer->form->addElement(form_makeField('text', 'data_edit[classes]', $data['classes'], $this->getLang('class'), 'data__classes'));
+
+            $el = new \dokuwiki\Form\InputElement('text', 'data_edit[classes]', $this->getLang('class'));
+            $el->id('data__classes');
+            $el->val($data['classes']);
+
+            $renderer->form->addElement($el);
```

Comment:
It is (now) possible to direct call `form->addTextInput()

For more examples see
https://www.dokuwiki.org/devel:form

some of the updates with new events have also bigger examples of the new form usage
https://www.patreon.com/posts/better-forms-58551930
(Please use code search)